arrows/vtk/mesh_coloration.h

```diff
@@ -0,0 +1,176 @@
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither the name Kitware, Inc. nor the names of any contributors may be
+ *    used to endorse or promote products derived from this software without
+ *    specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef KWIVER_ARROWS_VTK_MESH_COLORATION_H_
+#define KWIVER_ARROWS_VTK_MESH_COLORATION_H_
+
+// KWIVER includes
+#include <vital/algo/video_input.h>
+#include <vital/config/config_block_types.h>
+#include <vital/types/camera_map.h>
+#include <vital/types/camera_perspective.h>
+
+// VTK Class
+class vtkFloatArray;
+class vtkPolyData;
+class vtkRenderWindow;
+
+#include <arrows/vtk/kwiver_algo_vtk_export.h>
+#include <vtkSmartPointer.h>
+
+#include <string>
+#include <vector>
+
+namespace kwiver {
+namespace arrows {
+namespace vtk {
+
+/// Color a mesh from video, mask and cameras.
+class KWIVER_ALGO_VTK_EXPORT mesh_coloration : public kwiver::vital::noncopyable
+{
+
+public:
+  mesh_coloration();
+  /// Construct object to color a mesh
+  /**
+   * Input parameters are:
+   * \param video_config Configuration for reading the video
+   * \param video_path Video file path
+   * \param mask_config Configuration for reading the mask
+   * \param mask_path Mask file path
+   */
+  mesh_coloration(kwiver::vital::config_block_sptr& video_config,
+                  std::string const& video_path,
+                  kwiver::vital::config_block_sptr& mask_config,
+                  std::string const& mask_path,
+                  kwiver::vital::camera_map_sptr& cameras);
+
+  /// Input mesh to be colored
+  void set_input(vtkSmartPointer<vtkPolyData> input);
+  /// Input mesh to be colored
+  vtkSmartPointer<vtkPolyData> get_input();
+  /// Output mesh. Can be the same as the `input'
+  void set_output(vtkSmartPointer<vtkPolyData> mesh);
+  /// Output mesh. Can be the same as the `input'
+  vtkSmartPointer<vtkPolyData> get_output();
+  /// Average color from frames that satisfy frameIndex mod `sampling' == 0
+  void set_frame_sampling(int sampling);
+  /// Set color from `frameIndex'
+  void set_frame(int frameIndex)
```

Comment:
This one probably should be `frame_index`, but if this is the only issue, and the CI builds pass, we should merge this PR as is.  We could change this in your follow-on PR.